src/variables.cpp

```diff
@@ -9,47 +11,76 @@ VariablesAndValues *VariablesAndValues::getInstance()
 {
 	if (self == nullptr) {
 		self = new VariablesAndValues();
+
+		self->dataStorage = obs_data_create();
+		self->dataStoragePath = obs_module_config_path("variables.json");
+
+		auto *existingData = obs_data_create_from_json_file_safe(self->dataStoragePath, "bak");
+		obs_data_apply(self->dataStorage, existingData);
 	}
 	return self;
 }
 
+void VariablesAndValues::updateVariables(std::set<QString> updatedList) {
```

Comment:
would this be better as:

void VariablesAndValues::updateVariables(const std::set<QString> &updatedList) {

to avoid a copy construction?

Reply:
Absolutely.  I think my original intent was to copy, but now that it's been a bit it seems a bit silly to do that.  I've updated the code.